libraries/AP_Mount/AP_Mount_XFRobot.cpp

```diff
@@ -0,0 +1,590 @@
+        if (!_uart->read(b)) {
+            continue;
+        }
+
+        // add latest byte to buffer
+        msg_buff.bytes[msg_buff_len++] = b;
+
+        // protect against overly long messages
+        if (msg_buff_len > AP_MOUNT_XFROBOT_RECV_LENGTH_MAX) {
+            reset_parser = true;
+        }
+
+        // process byte depending upon current state
+        switch (parser.state) {
+        case ParseState::WAITING_FOR_HEADER1:
+            if (b == RECV_HEADER1) {
+                parser.state = ParseState::WAITING_FOR_HEADER2;
+            }
+            break;
+        case ParseState::WAITING_FOR_HEADER2:
+            if (b == RECV_HEADER2) {
+                parser.state = ParseState::WAITING_FOR_LENGTH_LOW;
+            } else {
+                reset_parser = true;
+            }
+            break;
+        case ParseState::WAITING_FOR_LENGTH_LOW:
+            parser.len_expected = b;
+            parser.state = ParseState::WAITING_FOR_LENGTH_HIGH;
+            break;
+        case ParseState::WAITING_FOR_LENGTH_HIGH:
+            parser.len_expected |= ((uint16_t)b << 8);
+            if (parser.len_expected < RECV_LENGTH_MIN || parser.len_expected > AP_MOUNT_XFROBOT_RECV_LENGTH_MAX) {
+                reset_parser = true;
+            } else {
+                parser.state = ParseState::WAITING_FOR_DATA;
+            }
+            break;
+        case ParseState::WAITING_FOR_DATA:
+            // check if we have received all data bytes.  the final two bytes are for the CRC
+            if (msg_buff_len >= parser.len_expected - 2) {
+                parser.state = ParseState::WAITING_FOR_CRC_HIGH;
+            }
+            break;
+        case ParseState::WAITING_FOR_CRC_HIGH:
+            parser.crc = (uint16_t)b << 8;
+            parser.state = ParseState::WAITING_FOR_CRC_LOW;
+            break;
+        case ParseState::WAITING_FOR_CRC_LOW:
+            parser.crc |= b;
+            const uint16_t expected_crc = crc16_ccitt(msg_buff.bytes, msg_buff_len - 2, 0);
+            if (expected_crc == parser.crc) {
+                // successfully received a message, do something with it
+                process_packet();
+            }
+            reset_parser = true;
+            break;
+        }
+
+        // handle reset of parser
+        if (reset_parser) {
+            msg_buff_len = 0;
+            parser.state = ParseState::WAITING_FOR_HEADER1;
+            reset_parser = false;
+        }
+    }
+}
+
+// process successfully decoded packets held in the msg_buff structure
+void AP_Mount_XFRobot::process_packet()
+{
+    // sanity check size of packet received
+    if (msg_buff_len < sizeof(GCUSimpleReply)) {
+        return;
+    }
+
+    // extract latest attitude
+    attitude_latest = {
+        .roll_ef_deg = msg_buff.simple_reply.main.roll_abs_cd * 0.01f,
+        .pitch_ef_deg = msg_buff.simple_reply.main.pitch_abs_cd * 0.01f,
+        .yaw_bf_deg = msg_buff.simple_reply.main.angle_z * 0.01f,
```

Comment:
I'm *assuming* this one is also in cd

Reply:
Hi @peterbarker,

Yes, it's centi-degrees in the datasheet and I've also tested that the angles appear correctly.
![image](https://github.com/user-attachments/assets/389f7244-74e8-4e25-a9cc-b7749896ddd8)

The reason we use "angle_z" (which is a body-frame yaw) instead of "yaw_abs_cd" is just that that's how we tend to report camera gimbal angles in AP.